src/main/scala/org/broadinstitute/dsde/workbench/sam/Boot.scala

```diff
@@ -117,32 +122,78 @@ object Boot extends IOApp with LazyLogging {
       (backgroundDirectoryDAO, backgroundAccessPolicyDAO, backgroundLdapExecutionContext, _) <- createDAOs(appConfig, DatabaseNames.Background, appConfig.directoryConfig.backgroundConnectionPoolSize, "background")
 
       blockingEc <- ExecutionContexts.fixedThreadPool[IO](24)
-      appDependencies <- appConfig.googleConfig match {
-        case Some(config) =>
-          for {
-            googleFire <- GoogleFirestoreInterpreter.firestore[IO](
-              config.googleServicesConfig.serviceAccountCredentialJson.firestoreServiceAccountJsonPath.asString)
-            googleStorage <- GoogleStorageInterpreter.storage[IO](
-              config.googleServicesConfig.serviceAccountCredentialJson.defaultServiceAccountJsonPath.asString)
-            googleKmsClient <- GoogleKmsInterpreter.client[IO](
-              config.googleServicesConfig.serviceAccountCredentialJson.defaultServiceAccountJsonPath.asString)
-          } yield {
-            val ioFireStore = GoogleFirestoreInterpreter[IO](googleFire)
-            // googleServicesConfig.resourceNamePrefix is an environment specific variable passed in https://github.com/broadinstitute/firecloud-develop/blob/fade9286ff0aec8449121ed201ebc44c8a4d57dd/run-context/fiab/configs/sam/docker-compose.yaml.ctmpl#L24
-            // Use resourceNamePrefix to avoid collision between different fiab environments (we share same firestore for fiabs)
-            val lock =
-              DistributedLock[IO](s"sam-${config.googleServicesConfig.resourceNamePrefix.getOrElse("local")}", appConfig.distributedLockConfig, ioFireStore)
-            val newGoogleStorage = GoogleStorageInterpreter[IO](googleStorage, blockingEc)
-            val googleKmsInterpreter = GoogleKmsInterpreter[IO](googleKmsClient, blockingEc)
-            val resourceTypeMap = appConfig.resourceTypes.map(rt => rt.name -> rt).toMap
-            val cloudExtension = createGoogleCloudExt(foregroundAccessPolicyDAO, foregroundDirectoryDAO, registrationDAO, config, resourceTypeMap, lock, newGoogleStorage, googleKmsInterpreter)
-            val googleGroupSynchronizer = new GoogleGroupSynchronizer(backgroundDirectoryDAO, backgroundAccessPolicyDAO, cloudExtension.googleDirectoryDAO, cloudExtension, resourceTypeMap)(backgroundLdapExecutionContext)
-            val cloudExtensionsInitializer = new GoogleExtensionsInitializer(cloudExtension, googleGroupSynchronizer)
-            createAppDepenciesWithSamRoutes(appConfig, cloudExtensionsInitializer, foregroundAccessPolicyDAO, foregroundDirectoryDAO, registrationDAO)
-          }
-        case None => cats.effect.Resource.pure[IO, AppDependencies](createAppDepenciesWithSamRoutes(appConfig, NoExtensionsInitializer, foregroundAccessPolicyDAO, foregroundDirectoryDAO, registrationDAO))
-      }
-    } yield appDependencies
```

Comment:
extracted this code to a function down below which returns `cloudExtensionsInitializer` and move the call to `createAppDepenciesWithSamRoutes` to the yield